### PR TITLE
Fix some Temporal issues

### PR DESCRIPTION
**Pull request by gergocs**
Signed-off-by: Gergo Csizi gergocs@inf.u-szeged.hu

### CODE REVIEW

**Comment by gergocs** [comment]
Patch is ready for review

File: src/builtins/BuiltinTemporal.cpp [hunk 1]
```diff
@@ -254,7 +256,8 @@ static Value builtinTemporalPlainTimeCompare(ExecutionState& state, Value thisVa
 static Value builtinTemporalPlainTimeCalendar(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
     checkTemporalPlainTime(state, thisValue);
-    return thisValue.asObject()->asTemporalPlainTimeObject()->getCalendar();
+    auto val = thisValue.asObject()->asTemporalPlainTimeObject()->getCalendar();
+    return Value(val);
```
Comment (zherczeg) [inline src/builtins/BuiltinTemporal.cpp:260]: `return Value(thisValue.asObject()->asTemporalPlainTimeObject()->getCalendar());` ?

File: src/builtins/BuiltinTemporal.cpp [hunk 2]
```diff
@@ -309,22 +312,26 @@ static Value builtinTemporalPlainTimeWith(ExecutionState& state, Value thisValue
...
 
     auto result = TemporalPlainTime::regulateTime(state,
-                                                  partialTime["hour"].isUndefined() ? partialTime["hour"].asInt32() : temporalTime->getHour(),
-                                                  partialTime["minute"].isUndefined() ? partialTime["minute"].asInt32() : temporalTime->getMinute(),
-                                                  partialTime["second"].isUndefined() ? partialTime["second"].asInt32() : temporalTime->getSecond(),
-                                                  partialTime["millisecond"].isUndefined() ? partialTime["millisecond"].asInt32() : temporalTime->getMillisecond(),
-                                                  partialTime["microsecond"].isUndefined() ? partialTime["microsecond"].asInt32() : temporalTime->getMicrosecond(),
-                                                  partialTime["nanosecond"].isUndefined() ? partialTime["nanosecond"].asInt32() : temporalTime->getNanosecond(),
+                                                  partialTime[HOUR_UNIT].isUndefined() ? partialTime[HOUR_UNIT].asInt32() : temporalTime->getHour(),
```
Comment (zherczeg) [inline src/builtins/BuiltinTemporal.cpp:315]: I like this change very much!

File: src/runtime/TemporalObject.cpp [hunk 3]
```diff
@@ -273,7 +271,7 @@ TemporalObject::DateTime TemporalObject::parseValidIso8601String(ExecutionState&
         }
 
         // Calendar
-        if (isoString.rfind("[u-ca=") == 0) {
+        if (isoString.rfind("[u-ca=") != std::string::npos) {
```
Comment (zherczeg) [inline src/runtime/TemporalObject.cpp:274]: Can it be anywhere in the string?

**Review by zherczeg — CHANGES_REQUESTED** [review]
Good patch, only a few changes.

File: src/runtime/TemporalObject.cpp (hunk 3 above)
Comment (gergocs) [reply src/runtime/TemporalObject.cpp:274]: No. It need to be started at index. I will fix it.

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**